File: app/javascript/models/Color.ts
export enum Color {
    Blue = 'blue',
    White = 'white',
    Red = 'red',
    Orange = 'orange',
    Teal = 'teal',
    Brown = 'brown',
    Pink = 'pink'
}

export function colorClassName(color: Color) {
    return color.toString()
}

File: app/models/color.rb
class Color < ActiveHash::Base
    include ActiveHash::Enum
    enum_accessor :name
    self.data = [
        { id: 'blue' },
        { id: 'white' },
        { id: 'red' },
        { id: 'orange' },
        { id: 'teal' },
        { id: 'brown' },
        { id: 'pink' }
    ]
end
